### PR TITLE
Bugfix/118 create dirs on run

### LINKED ISSUE #118: [BUG] on first run, `cucumber_output` is not writable and `audit_output` does not exist
**Describe the bug**
On first run, using defaults, Probr expects to be able to write to directories `cucumber_output` and `audit_output`. There are two problems with this:

1. `cucumber_output` is created with `d---------`, which is insufficient for write
1. `audit_output` is not created at al

**To Reproduce**
Make no changes to the config, build, and run:

1. Compile: `go build -o probr cmd/main.go`
1. Run: `./probr`

**Expected behavior**
First launch experience should be clean - Probr should work with no other configuration.

**Screenshots**
Not applicable.

**Additional context**
Bug exists on Mac. Would appreciate a Windows user double checking just in case the fix doesn't work.

### PR DESCRIPTION
**Pull request by paulm-jones**
This Pull Request fixes two issues with the defaults we use for outputs.

With no additional config, we try to use `cucumber_output` and `audit_output`. The former is created, but isn't writable. The latter is not created at all.

This change:

- Changes the `cucumber_output` file mode to 0755
- Creates an accessor for `AuditDir` which will create the directory if it doesn't exist. This config value is only read in one place - we replace it with a call to the accessor.

Arguably I think the `os.Mkdir` in the `getOutputPath` helper method should also behave in the same way - we should move the Mkdir into an accessor as well.

closes #118

### CODE REVIEW

File: probr.go [hunk 1]
```diff
@@ -20,33 +20,26 @@ func RunAllProbes() (int, *coreengine.ProbeStore, error) {
...
 
-//GetAllProbeResults ...
-func GetAllProbeResults(ps *coreengine.ProbeStore) (map[string]string, error) {
+//GetAllProbeResults maps ProbeStore results to strings
+func GetAllProbeResults(ps *coreengine.ProbeStore) map[string]string {
 	out := make(map[string]string)
 	for name := range ps.Probes {
-		r, n, err := ReadProbeResults(ps, name)
+		results, name, err := readProbeResults(ps, name)
 		if err != nil {
```
Comment (paulm-jones) [inline probr.go:28]: I think this was a bug.

We try to read all of the Probe results, and make the judgement that if there are any failures in reading them from `coreengine.ProbeStore` then it likely indicates that the `cucumber_output` folder is unavailable.

On first run, I was seeing partial failures - files were in `cucumber_output` just fine, but one was failing (the first, I think), and so we were incorrectly judging it to be a complete failure.

For now I am just putting the error string we generate ("test with name X not found") in the output map in place of the actual results. I think this is fine, as the only thing we do with the output is check the output map is of non-zero length and not nil.

File: probr.go [hunk 2]
```diff
@@ -20,33 +20,26 @@ func RunAllProbes() (int, *coreengine.ProbeStore, error) {
...
+			out[name] = results
 		}
 	}
-	return out, nil
+	return out
 }
 
-//ReadProbeResults ...
-func ReadProbeResults(ps *coreengine.ProbeStore, name string) (string, string, error) {
+func readProbeResults(ps *coreengine.ProbeStore, name string) (probeResults, probeName string, err error) {
```
Comment (paulm-jones) [inline probr.go:37]: I think this reads better with named outputs.

File: cmd/main.go [hunk 3]
```diff
@@ -60,7 +56,7 @@ func main() {
 }
 
 func showIndicator() bool {
-	return (config.Vars.LogLevel == "ERROR" && !config.Vars.Silent)
```
Comment (paulm-jones) [inline cmd/main.go:63]: Go style thing - doesn't need wrapping in brackets.

File: go.mod [hunk 4]
```diff
@@ -12,11 +12,11 @@ require (
 	github.com/cucumber/messages-go/v10 v10.0.3
 	github.com/gobuffalo/packr/v2 v2.5.1
 	github.com/golang/protobuf v1.3.3 // indirect
+	github.com/google/gops v0.3.13 // indirect
```
Comment (paulm-jones) [inline go.mod:15]: for the debugger

File: internal/config/config.go [hunk 5]
```diff
@@ -161,3 +161,8 @@ func LogConfigState() {
 	s, _ := json.MarshalIndent(Vars, "", "  ")
 	log.Printf("[NOTICE] Config State: %s", s)
 }
+
+func AuditDir() string {
```
Comment (paulm-jones) [inline internal/config/config.go:165]: Create this directory on read if it doesn't exist, ignoring the error if it does.

File: internal/coreengine/probe_helpers.go [hunk 6]
```diff
@@ -47,7 +47,7 @@ func getRootDir() (string, error) {
 // plus the test name supplied
 func getOutputPath(t string) (*os.File, error) {
 
-	os.Mkdir(config.Vars.CucumberDir, os.ModeDir)
+	_ = os.Mkdir(config.Vars.CucumberDir, 0755)
```
Comment (paulm-jones) [inline internal/coreengine/probe_helpers.go:50]: I think arguably could pull this into same place as `AuditDir`.

**Review by paulm-jones — COMMENTED** [review]
@eddie-knight please have a read

**Review by eddie-knight — APPROVED** [review]
Tested it out, looks good across the board. Thanks for taking care of this!

**Pull request merged**